chalice/deploy/planner.py

```diff
@@ -337,6 +365,100 @@ def _plan_managediamrole(self, resource):
             )
         ]
 
+    def _plan_snslambdasubscription(self, resource):
+        # type: (models.SNSLambdaSubscription) -> Sequence[_INSTRUCTION_MSG]
+        function_arn = Variable(
+            '%s_lambda_arn' % resource.lambda_function.resource_name
+        )
+        topic_arn_varname = '%s_topic_arn' % resource.resource_name
+        subscribe_varname = '%s_subscription_arn' % resource.resource_name
+        # To keep the user API simple, we only require the topic
+        # name and not the ARN.  However, the APIs require the topic
+        # ARN so we need to reconstruct it here in the planner.
+        instruction_for_topic_arn = [
+            models.BuiltinFunction(
+                'parse_arn',
+                [function_arn],
+                output_var='parsed_lambda_arn',
+            ),
+            models.JPSearch('account_id',
+                            input_var='parsed_lambda_arn',
+                            output_var='account_id'),
+            models.JPSearch('region',
+                            input_var='parsed_lambda_arn',
+                            output_var='region_name'),
+            models.StoreValue(
+                name=topic_arn_varname,
+                value=StringFormat(
+                    'arn:aws:sns:{region_name}:{account_id}:%s' % (
+                        resource.topic
+                    ),
+                    ['region_name', 'account_id'],
+                ),
+            ),
+        ]  # type: List[_INSTRUCTION_MSG]
```

Comment:
Does this require the type annotation?

Reply:
Yes, otherwise it infers the type based on the elements in the list, and it will incorrectly infer these all to be `Instruction` types because we're not using the tuple form that could also include an optional message we want displayed.